### PR TITLE
frameTransformServer: display transform tree 

### PR DESCRIPTION
**Pull request by randaz81**
frameTransformServer: added a new rpc command 'generate_view' which generates via dot (graphviz) a `frames.pdf` file, containing a diagram of the transforms tree.

You can use a pdf viewer, i.e. `evince frames.pdf` to display the transforms tree and obtain a plot such as:
image

### CODE REVIEW

File: src/devices/transformServer/FrameTransformServer.cpp [hunk 1]
```diff
@@ -224,6 +224,99 @@ void FrameTransformServer::list_response(yarp::os::Bottle& out)
...
+            <tr><td port = \"i3\">&nbsp;</td></tr>\n\
+            <tr><td port = \"i4\">&nbsp;</td></tr>\n\
+            </table>>]\n\
+          key:i1:e -> key2:i1:w [color = blue]\n\
+          key:i2:e -> key2:i2:w [color = blue, style=dashed]\n\
+          key:i3:e -> key2:i3:w [color = black]\n\
+          key:i4:e -> key2:i4:w [color = black, style=dashed]\n\
+        } }";
+
+    string command_string = "printf '"+dot_string+ legend + "' | dot -Tpdf > frames.pdf";
```
Comment (drdanz) [inline src/devices/transformServer/FrameTransformServer.cpp:297]: Codacy found an issue: [Variable 'command_string' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5969044)

File: src/devices/transformServer/FrameTransformServer.cpp [hunk 2]
```diff
@@ -224,6 +224,99 @@ void FrameTransformServer::list_response(yarp::os::Bottle& out)
     }
 }
 
+string FrameTransformServer::get_matrix_as_text(Transforms_server_storage* storage, int i)
+{
+    //add here the latex to code to display the trnasformation matrix
+    //@@@TO BE COMPLETED
+    string s = "\\begin{ bmatrix } \
```
Comment (drdanz) [inline src/devices/transformServer/FrameTransformServer.cpp:231]: Codacy found an issue: [Variable 's' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5969044)

File: src/devices/transformServer/FrameTransformServer.cpp [hunk 3]
```diff
@@ -224,6 +227,118 @@ void FrameTransformServer::list_response(yarp::os::Bottle& out)
...
+            <tr><td port = \"i3\">&nbsp;</td></tr>\n\
+            <tr><td port = \"i4\">&nbsp;</td></tr>\n\
+            </table>>]\n\
+          key:i1:e -> key2:i1:w [color = blue]\n\
+          key:i2:e -> key2:i2:w [color = blue, style=dashed]\n\
+          key:i3:e -> key2:i3:w [color = black]\n\
+          key:i4:e -> key2:i4:w [color = black, style=dashed]\n\
+        } }";
+
+    string command_string = "printf '"+dot_string+ legend + "' | dot -Tpdf > frames.pdf";
```
Comment (drdanz) [inline src/devices/transformServer/FrameTransformServer.cpp:319]: Codacy found an issue: [Variable 'command_string' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5969044)

**Comment by Nicogene** [comment]
cc @aikolina

**Review by drdanz — APPROVED** [review]

**Pull request merged**